src/main/java/co/rsk/federate/signing/hsm/advanceblockchain/HSMBookkeepingService.java

```diff
@@ -134,11 +135,13 @@ protected void informConfirmedBlockHeaders() {
         informing = true;
         logger.info("Starting HSM bookkeeping process");
         try {
-            Block hsmCurrentBestBlock = getHsmBestBlock();
             if (hsmCurrentBestBlock == null) {
-                logger.error("[informConfirmedBlockHeaders] Can't found HSM Best Block in blockStore.");
-                informing = false;
-                return;
+                hsmCurrentBestBlock = getHsmBestBlock();
+                if (hsmCurrentBestBlock == null) {
```

Comment:
Under what circumstance would the HSM best block not be found in the block store?

That seems like a ver abnormal situation, perhaps we could do this validation inside `getHsmBestBlock` method and if not found then throw an exception

Reply:
Updated